### PR TITLE
Generate open graph data for each article

### PR DESCRIPTION
**Pull request by commuterjoy**
Generates basic open graph data.

See also - https://github.com/Financial-Times/next-express/pull/191

```
<meta property="og:title" content="Monitise issues fresh revenue warning as model evolves"/>
<meta property="og:url" content="https://next.ft.com/0786d5a8-23b2-11e5-bd83-71cb60e8f08c"/>
<meta property="og:description" content="Mobile money group predicts return to profit by 2016"/>
<meta property="og:image" content="http://com.ft.imagepublish.prod.s3.amazonaws.com/39278302-23ba-11e5-bd83-71cb60e8f08c"/>
<meta property="og:type" content="article"/>
<meta property="og:site_name" content="Financial Times"/>
```

### CODE REVIEW

File: server/controllers/article.js [hunk 1]
```diff
@@ -40,6 +41,25 @@ module.exports = function(req, res, next) {
 		useElasticSearch: res.locals.flags.elasticSearchItemGet
 	});
 
+	var mainImage = function (articleV2) {
+		
+		if (!articleV2.mainImage) {
+			return Promise.resolve();
+		}
+		
+		if (!res.locals.flags.openGraph) {
```
Comment (commuterjoy) [inline server/controllers/article.js]: maybe this should be somewhere else.

File: server/controllers/article.js [hunk 2]
```diff
@@ -40,6 +41,25 @@ module.exports = function(req, res, next) {
...
+			return Promise.resolve();
+		} 
+
+		return api.content({ uuid: extractUuid(articleV2.mainImage.id), type: 'ImageSet' })
+			.then(function (images) {
+				var image = images.members.reduce(function (a, b) {
+					return a;
+				});
+				return api.content({ uuid: extractUuid(image.id), type: 'ImageSet' });  
+			});
```
Comment (commuterjoy) [inline server/controllers/article.js]: two extra requests for every article with a main image :/

File: server/utils/open-graph.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+
+var extractUuid = require('./extract-uuid');
+
+module.exports = function (article, articleV1, mainImage) {
+	return {
+		title: article.title,
+		description: articleV1.editorial.standFirst,
```
Comment (commuterjoy) [inline server/utils/open-graph.js]: v1 dependency :/

File: server/controllers/article.js [hunk 4]
```diff
@@ -40,6 +41,25 @@ module.exports = function(req, res, next) {
...
+			return Promise.resolve();
+		}
+		
+		if (!res.locals.flags.openGraph) {
+			return Promise.resolve();
+		} 
+
+		return api.content({ uuid: extractUuid(articleV2.mainImage.id), type: 'ImageSet' })
+			.then(function (images) {
+				var image = images.members.reduce(function (a, b) {
```
Comment (commuterjoy) [inline server/controllers/article.js]: @matthew-andrews uncertain of how to process `images.members` - so just return the first item.

File: server/utils/open-graph.js (hunk 3 above)
Comment (matthew-andrews) [inline server/utils/open-graph.js]: coming this quarter! :-)

File: server/controllers/article.js (hunk 4 above)
Comment (matthew-andrews) [inline server/controllers/article.js]: your guess is as good as mine.  undocumented so whatever……

File: server/controllers/article.js (hunk 2 above)
Comment (matthew-andrews) [inline server/controllers/article.js]: i think we actually skip the second request becasue the binary url can be derived from the uuid……

e.g. uuid of the 2nd request could be 8cd0b1a2-ed57-11e4-bebf-00144feab7de …… which is http://com.ft.imagepublish.prod.s3.amazonaws.com/8cd0b1a2-ed57-11e4-bebf-00144feab7de

kind of naughty of us though…… and they could change it at anytime and break us…… and the image service!

**Pull request merged**